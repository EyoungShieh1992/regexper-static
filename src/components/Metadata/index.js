import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { Helmet } from 'react-helmet';

const Metadata = ({ title, description }) => {
  const { i18n } = useTranslation();
  const helmetProps = {
    title: title ? `Regexper - ${ title }` : 'Regexper',
    htmlAttributes: {
      lang: i18n.language
    },
    meta: [
      {
        name: 'description',
        content: description
      }
    ]
  };

  return <Helmet { ...helmetProps }></Helmet>;
};

Metadata.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string
};

export default Metadata;
