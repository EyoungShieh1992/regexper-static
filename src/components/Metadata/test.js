jest.mock('react-helmet', () => {
  const helmet = jest.requireActual('react-helmet');
  return {
    ...helmet,
    Helmet: require('__mocks__/component-mock').buildMock(helmet.Helmet)
  };
});

import React from 'react';
import { render } from 'react-testing-library';

import Metadata from 'components/Metadata';

describe('Metadata', () => {
  test('rendering', () => {
    const { asFragment } = render(
      <Metadata/>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('rendering with a title and description', () => {
    const { asFragment } = render(
      <Metadata
        title="Testing"
        description="Test description" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
