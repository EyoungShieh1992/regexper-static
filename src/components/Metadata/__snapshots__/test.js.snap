// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metadata rendering 1`] = `
<DocumentFragment>
  <span
    data-component="HelmetWrapper"
    data-props="{
  \\"title\\": \\"Regexper\\",
  \\"htmlAttributes\\": {},
  \\"meta\\": [
    {
      \\"name\\": \\"description\\"
    }
  ]
}"
  />
</DocumentFragment>
`;

exports[`Metadata rendering with a title and description 1`] = `
<DocumentFragment>
  <span
    data-component="HelmetWrapper"
    data-props="{
  \\"title\\": \\"Regexper - Testing\\",
  \\"htmlAttributes\\": {},
  \\"meta\\": [
    {
      \\"name\\": \\"description\\",
      \\"content\\": \\"Test description\\"
    }
  ]
}"
  />
</DocumentFragment>
`;
